cinema/views.py

```diff
@@ -1 +1,68 @@
-# write views here
+from rest_framework import viewsets
+
+from .models import Genre, Actor, CinemaHall, Movie, MovieSession
+from .serializers import (
+    GenreSerializer,
+    GenreListSerializer,
+    ActorSerializer,
+    ActorListSerializer,
+    CinemaHallSerializer,
+    MovieSerializer,
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieSessionDetailSerializer,
+)
+
+
+class GenresViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return GenreListSerializer
+        return GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return ActorListSerializer
+        return ActorSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all().prefetch_related("genres", "actors")
+    serializer_class = MovieSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all().prefetch_related(
+        "movie",
+        "cinema_hall"
+    )
```

Comment:
no need to use `all()` mehtod before joins
also are u sure about prefetch_related for moviesessions ?